services/deposit.py

```diff
@@ -1,41 +1,40 @@
+import datetime
+from typing import List
 from sqlalchemy import select
-
-from db import async_session_maker
+from db import session_execute, session_commit, session_refresh, get_db_session
 from models.deposit import Deposit
 
 
 class DepositService:
 
-    @staticmethod
-    async def get_next_user_id() -> int:
-        async with async_session_maker() as session:
-            query = select(Deposit.id).order_by(Deposit.id.desc()).limit(1)
-            last_user_id = await session.execute(query)
-            last_user_id = last_user_id.scalar()
-            if last_user_id is None:
-                return 0
-            else:
-                return int(last_user_id) + 1
-
     @staticmethod
     async def create(tx_id, user_id, network, token_name, amount, vout=None):
-        async with async_session_maker() as session:
-            next_deposit_id = await DepositService.get_next_user_id()
-            dep = Deposit(id=next_deposit_id,
-                          user_id=user_id,
+        async with get_db_session() as session:
+            dep = Deposit(user_id=user_id,
                           tx_id=tx_id,
                           network=network,
                           token_name=token_name,
                           amount=amount,
                           vout=vout)
             session.add(dep)
-            await session.commit()
-            return next_deposit_id
+            await session_commit(session)
+            await session_refresh(session, dep)
+            return dep.id
 
     @staticmethod
-    async def get_by_user_id(user_id):
-        async with async_session_maker() as session:
+    async def get_by_user_id(user_id: int):
+        async with get_db_session() as session:
             stmt = select(Deposit).where(Deposit.user_id == user_id)
-            deposits = await session.execute(stmt)
+            deposits = await session_execute(stmt, session)
             deposits = deposits.scalars().all()
             return deposits
+
+    @staticmethod
+    async def get_by_timedelta(timedelta_int: int) -> List[Deposit]:
+        current_time = datetime.datetime.now()
+        timedelta = datetime.timedelta(days=int(timedelta_int))
+        time_to_subtract = current_time - timedelta
+        async with get_db_session() as session:
+            stmt = select(Deposit).where(Deposit.deposit_datetime >= time_to_subtract)
+            deposits = await session_execute(stmt, session)
+            return deposits.scalars().all()
```

Comment:
would this contribute to #67 ?

Reply:
This functionality will be developed in a separate feature branch